### PR TITLE
Improve error handling for legacy getStats

### PR DESCRIPTION
**Pull request by yiming-amzn**
*Issue #, if available:*
https://github.com/aws/connect-rtc-js/issues/57
*Description of changes:*
Before the change, in the legacy getStats function callback, we are throwing an error which streamJS does not catch.
After the change, we are return the promise with error message in the reject function, so that streamJS can catch it and log it in debug level
See details here: https://stackoverflow.com/a/33446005
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/js/rtc_session.js [hunk 1]
```diff
@@ -946,11 +946,11 @@ export default class RtcSession {
                 // get legacy stats report as a promise
                 if (this._legacyStatsReportSupport) {
                     var self = this;
-                    return new Promise(function(resolve) {
+                    return new Promise(function(resolve, reject) {
```
Comment (Vignendra) [inline src/js/rtc_session.js:949]: Can you add a corresponding unit test?

File: package.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
     "version": "grunt build && grunt copyForPublish && git add package.json release/* ",
     "postversion": "export GITTAG=\"echo $(git describe --abbrev=0 --tags | sed 's/^v//')\" && git push --force --set-upstream origin bumpVersion --follow-tags && git checkout gh-pages && git pull && cp out/connect-rtc-debug.js ./connect-rtc-debug-`$GITTAG`.js && cp out/connect-rtc.min.js ./connect-rtc-`$GITTAG`.min.js && ln -fs connect-rtc-debug-`$GITTAG`.js connect-rtc-debug-latest.js && ln -fs connect-rtc-`$GITTAG`.min.js connect-rtc-latest.min.js && git add connect-rtc*.js && git commit -m `$GITTAG` && git push --set-upstream origin gh-pages && git checkout master",
     "prepublish": "grunt build",
-    "test": "grunt && mocha --compilers js:babel-core/register test/unit"
+    "test": "grunt && nyc --reporter=html mocha --compilers js:babel-core/register test/unit"
```
Comment (Vignendra) [inline package.json:26]: Do we need to add this to any of the above scripts? Right now I see it added only to preversion script as dependent build script that needs to be run.

Comment (Vignendra) [reply package.json:26]: Discussed offline.

Comment (yiming-amzn) [reply package.json:26]: >If preversion, version, or postversion are in the scripts property of the package.json, they will be executed as part of running npm version.

Discussed offline, npm version will call preversion by default so this line will be executed when we run npm version

File: test/unit/rtc_session.js [hunk 3]
```diff
@@ -760,4 +760,52 @@ describe('RTC session', () => {
...
+            clock = sinon.useFakeTimers();
+        });
+
+        afterEach(() => {
+            clock.restore();
+        });
+
+        it('for legacy stats, when digested stats is empty, error is handled', () => {
+            var errorCatched = false;
+            session.getRemoteAudioStats().then(function () {
```
Comment (yiming-amzn) [inline test/unit/rtc_session.js:801]: This unit test is to make sure the callback in session._pc.getStats() will not throw an error(old behavior).
See detailed explanation about the bug here: https://stackoverflow.com/questions/33445415/javascript-promises-reject-vs-throw/33446005#33446005

getRemoteAudioStats will invoke, `async getStats() ` and call session._pc.getStats(). session._pc.getStats(). is mocked method that will just invoke the callback function after a delay. This means that the unit test will invoke the callback function and verified that the call back function no longer throw the error

**Review by siwei-w — APPROVED** [review]

**Pull request merged**